modules/oe_theme_helper/tests/src/Functional/Plugin/field_group/InPageNavigationTest.php

```diff
@@ -0,0 +1,258 @@
+        ],
+      ];
+      $display->setComponent($field_name, $display_options);
+    }
+
+    $display->save();
+    $this->node = $this->drupalCreateNode($node_values);
+  }
+
+  /**
+   * Tests the in-page navigation field group formatters.
+   *
+   * Structure of the content:
+   * Field group "In-page navigation group"
+   * - Field group "In-page navigation item"
+   * -- Field 0
+   * -- Field 1
+   * - Field group "In-page navigation item"
+   * -- Single text extra field
+   * -- Multiple items extra field
+   * - Field group "In-page navigation item"
+   * -- Field group "Html elements"
+   * --- Test field 2
+   * --- Test field 3
+   * - Field group "In-page navigation item"
+   * -- Formatted extra field without content.
+   */
+  public function testOutput(): void {
+    // Create "Html element" field group.
+    $children_group_html_element = [
+      'field_test_2',
+      'field_test_3',
+    ];
+    $group_html_element = $this->createFieldGroup('html_element', 'html_element', $children_group_html_element, 10, ['show_label' => TRUE]);
+
+    // Create "In-page navigation items" field groups.
+    $children_group_inpage_nav_items = [
+      [
+        'field_test_0',
+        'field_test_1',
+      ],
+      [
+        'extra_field_single_text_test',
+        'extra_field_multiple_text_test',
+      ],
+      [
+        $group_html_element->group_name,
+      ],
+      // It is empty extra field, so field group shouldn't be rendered.
+      [
+        'extra_field_empty_formatted_test',
+      ],
+    ];
+    $groups_inpage_nav_item = [];
+    $children_group_inpage_nav = [];
+    for ($i = 0; $i < 4; $i++) {
+      $group = $this->createFieldGroup("inpage_nav_item_$i", 'oe_theme_helper_in_page_navigation_item', $children_group_inpage_nav_items[$i], $i);
+      $groups_inpage_nav_item[] = $group;
+      $children_group_inpage_nav[] = $group->group_name;
+    }
+
+    // Create In-page navigation group.
+    $this->createFieldGroup('main', 'oe_theme_helper_in_page_navigation', $children_group_inpage_nav);
+    $this->drupalGet('node/' . $this->node->id());
+
+    // Assert navigation part.
+    $this->assertSession()->elementTextContains('css', '.ecl-inpage-navigation__title', 'Field group main');
+    $navigation_list = $this->assertSession()->elementExists('css', '.ecl-inpage-navigation__list');
+    $navigation_list_items = $navigation_list->findAll('css', '.ecl-inpage-navigation__item');
+    $this->assertCount(3, $navigation_list_items);
+    for ($i = 0; $i < 3; $i++) {
+      $navigation_list_item_link = $navigation_list_items[$i]->find('css', '.ecl-link');
+      $this->assertEquals("Field group inpage_nav_item_$i", $navigation_list_item_link->getText());
+      $this->assertEquals("#field-group-inpage-nav-item-$i", $navigation_list_item_link->getAttribute('href'));
+    }
+
+    // Assert content part.
+    $wrapper = $this->assertSession()->elementExists('css', '.ecl-row.ecl-u-mt-l');
```

Comment:
This wrapper contains both the navigation and the content, we can declare it earlier and use it to assert that both navigation and content are inside it.